optimum/exporters/features.py

```diff
@@ -0,0 +1,744 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Model export features manager."""
+
+import importlib
+import os
+from functools import partial, reduce
+from typing import TYPE_CHECKING, Callable, Dict, Optional, Tuple, Type, Union
+
+from transformers import PretrainedConfig, is_tf_available, is_torch_available
+from transformers.utils import TF2_WEIGHTS_NAME, WEIGHTS_NAME, logging
+
+
+if TYPE_CHECKING:
+    from transformers import PreTrainedModel, TFPreTrainedModel
+
+    from .base import ExportConfig
+
+
+logger = logging.get_logger(__name__)  # pylint: disable=invalid-name
+
+if is_torch_available():
+    from transformers.models.auto import (
+        AutoModel,
+        AutoModelForCausalLM,
+        AutoModelForImageClassification,
+        AutoModelForImageSegmentation,
+        AutoModelForMaskedImageModeling,
+        AutoModelForMaskedLM,
+        AutoModelForMultipleChoice,
+        AutoModelForObjectDetection,
+        AutoModelForQuestionAnswering,
+        AutoModelForSemanticSegmentation,
+        AutoModelForSeq2SeqLM,
+        AutoModelForSequenceClassification,
+        AutoModelForTokenClassification,
+    )
+if is_tf_available():
+    from transformers.models.auto import (
+        TFAutoModel,
+        TFAutoModelForCausalLM,
+        TFAutoModelForMaskedLM,
+        TFAutoModelForMultipleChoice,
+        TFAutoModelForQuestionAnswering,
+        TFAutoModelForSemanticSegmentation,
+        TFAutoModelForSeq2SeqLM,
+        TFAutoModelForSequenceClassification,
+        TFAutoModelForTokenClassification,
+    )
+if not is_torch_available() and not is_tf_available():
+    logger.warning(
+        "The export features are only supported for PyTorch or TensorFlow. You will not be able to export models"
+        " without one of these libraries installed."
+    )
+
+ExportConfigConstructor = Callable[[PretrainedConfig], "ExportConfig"]
+FeatureNameToExportConfigDict = Dict[str, ExportConfigConstructor]
+
+
+def supported_features_mapping(*supported_features: str, **exporters: str) -> Dict[str, FeatureNameToExportConfigDict]:
+    """
+    Generates the mapping between supported the features and their corresponding `ExportConfig` for a given model, for
+    every backend.
+
+    Args:
+        supported_features (`Tuple[str]`):
+        The names of the supported features.
+        exporters (`Dict[str, str]`):
+            The export backend name -> config class name mapping. For instance, for instance:
+            ```
```

Comment:
The syntax for docstring examples to render nice on hf.co/docs is to use something `>>>` and the `python` marker as follows:

    ```python
    >>> from transformers import Wav2Vec2Processor, Wav2Vec2ForCTC
    >>> from datasets import load_dataset
    >>> import torch
    ```